### PR TITLE
[KYUUBI #5308] Ensure release using Java 8

### PR DESCRIPTION
**Pull request by wForget**
### _Why are the changes needed?_

Avoid releasing new version with incorrect java version.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_

No.

### CODE REVIEW

File: build/release/release.sh [hunk 1]
```diff
@@ -52,6 +52,20 @@ if [[ ${RELEASE_VERSION} =~ .*-SNAPSHOT ]]; then
   exit 1
 fi
 
+if [[ -z ${JAVA_HOME} ]]; then
+  if [[ $(command -v java) ]]; then
+    # shellcheck disable=SC2046
+    export JAVA_HOME="$(dirname $(dirname $(which java)))"
```
Comment (pan3793) [inline build/release/release.sh:58]: This silently affects all following procedures, we only need to find the right `java`, but won't want to set `JAVA_HOME`.

Comment (pan3793) [reply build/release/release.sh:58]: assume there is a `java` at `/bin/java`, then the evaluated `JAVA_HOME` is `/`, does it make sense?

Comment (wForget) [reply build/release/release.sh:58]: How about just getting javacmd?
```
if [ -z "$JAVA_HOME" ] ; then
  JAVA_CMD="`\\unset -f command; \\command -v java`"
else
  JAVA_CMD="$JAVA_HOME/bin/java"
fi
```

Comment (pan3793) [reply build/release/release.sh:58]:
```
# Find the java binary
if [ -n "${JAVA_HOME}" ]; then
  JAVA="${JAVA_HOME}/bin/java"
else if [ "$(command -v java)" ]; then
  JAVA="java"
else
  echo "JAVA_HOME is not set" >&2
  exit 1
fi
```

Comment (wForget) [reply build/release/release.sh]: Thanks, fixed.

File: build/release/release.sh [hunk 2]
```diff
@@ -52,6 +52,23 @@ if [[ ${RELEASE_VERSION} =~ .*-SNAPSHOT ]]; then
...
+  echo "JAVA_HOME is not set" >&2
+  exit 1
+fi
+
+# Required java 1.8 version
+JAVA_VERSION=$($JAVA -version 2>&1 | awk -F '"' '/version/ {print $2}')
+if [[ $JAVA_VERSION != 1.8.* ]]; then
+  echo "Java version $JAVA_VERSION is not required 1.8"
+  exit 1
+fi
```
Comment (pan3793) [inline build/release/release.sh:70]:
```suggestion
if [ -n "${JAVA_HOME}" ]; then
  JAVA="${JAVA_HOME}/bin/java"
elif [ "$(command -v java)" ]; then
  JAVA="java"
else
  echo "JAVA_HOME is not set" >&2
  exit 1
fi

JAVA_VERSION=$($JAVA -version 2>&1 | awk -F '"' '/version/ {print $2}')
if [[ $JAVA_VERSION != 1.8.* ]]; then
  echo "Unexpected Java version: $JAVA_VERSION. Java 8 is required for release."
  exit 1
fi
```

**Review by pan3793 — APPROVED** [review]
I verified this patch locally, LGTM

**Comment by pan3793** [comment]
Also cc @cfmcgrady @cxzl25

**Review by cfmcgrady — APPROVED** [review]
LGTM

**Comment by pan3793** [comment]
Thanks, merged to master/1.8/1.7

**Pull request closed without merging**